Add static pod control plane / node bootstrap to arch guide

adellape opened the pull request:
http://file.rdu.redhat.com/~adellape/071118/staticpod310/architecture/infrastructure_components/kubernetes_infrastructure.html#control-plane-static-pods

http://file.rdu.redhat.com/~adellape/071118/staticpod310/architecture/infrastructure_components/kubernetes_infrastructure.html#node-bootstrapping

adellape commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Document: Document deployment topology changes in openshift](https://trello.com/c/GrTn5IlB/905-document-document-deployment-topology-changes-in-openshift)

adellape commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Document: Run control plane as static pod and enable node bootstrapping by default](https://trello.com/c/jyhJgKRR/848-document-run-control-plane-as-static-pod-and-enable-node-bootstrapping-by-default)

adellape commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Document: Run control plane as static pod and enable node bootstrapping by default](https://trello.com/c/jyhJgKRR/848-document-run-control-plane-as-static-pod-and-enable-node-bootstrapping-by-default)

mfojtik commented:
/lgtm

derekwaynecarr commented on architecture/infrastructure_components/kubernetes_infrastructure.adoc:
```diff
@@ -145,26 +189,26 @@ etcd.
-xref:master[master]. Nodes also have the required services to run pods,
-including the Docker service, a xref:kubelet[kubelet], and a xref:service-proxy[service proxy].
+Kubernetes cluster has the required services to be managed by the master. Nodes
+also have the required services to run pods, including the *docker* service, a
```
rather than say the docker service, can we just say including the container runtime?

derekwaynecarr commented:
@mfojtik i think we should describe how to do some basic actions (start,stop,restart) of the control plane components in order to ease transition.

xingxingxia commented on architecture/infrastructure_components/kubernetes_infrastructure.adoc:
```diff
@@ -74,6 +73,51 @@ use the `native` method but pre-configure your own load balancer of choice.
+Starting in {product-title} 3.10, the deployment model for installing and
+operating the core control plane components changed. Prior to 3.10, the API server
+and the controller manager components ran as stand-alone host processes operated
+by SystemD. In 3.10, these two components are moved to
```
Seems should type 'systemd' instead of 'SystemD' here and in other places? See https://www.freedesktop.org/wiki/Software/systemd/ 'Spelling' part.

adellape replied:
Had not seen that. That's great, thanks!

xingxingxia commented on architecture/infrastructure_components/kubernetes_infrastructure.adoc:
```diff
@@ -74,6 +73,51 @@ use the `native` method but pre-configure your own load balancer of choice.
+operations you can do on them is limited. For example:
+
+----
+$ oc get logs master-api-<hostname> -n kube-system
```
```
# oc get logs master-api-qe-geliumaster-etcd-1 -n kube-system
error: the server doesn't have a resource type "logs"
```
Should remove `get`, just `oc logs`

adellape commented:
@derekwaynecarr @mfojtik Added "Restarting Master Services" and "Viewing Master Service Logs" subsections, and moved the mirror pod discussion to its own subsection:

http://file.rdu.redhat.com/~adellape/071118/staticpod310/architecture/infrastructure_components/kubernetes_infrastructure.html#control-plane-static-pods

Need info on moving pod definitions for starting/stopping services.

adellape commented:
@aveshagarwal Added the node bootstrapping diagram, as well as the "Node Configuration Workflow" and "Modifying Node Configurations" subsections:

http://file.rdu.redhat.com/~adellape/071118/staticpod310/architecture/infrastructure_components/kubernetes_infrastructure.html#node-bootstrapping

adellape commented:
Pushed a few tweaks:

- Joined some of the mirror pod info together and moved the "Mirror Pods" subsection up.
- Added some ifdef's so that all of the super detailed info doesn't show up in the Dedicated and Online distros.

adellape commented:
/cherrypick enterprise-3.10
